File: packages/device-info/android/src/main/java/io/capawesome/capacitorjs/plugins/deviceinfo/classes/results/GetUptimeResult.java
package io.capawesome.capacitorjs.plugins.deviceinfo.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.deviceinfo.interfaces.Result;

public class GetUptimeResult implements Result {

    private final long uptime;

    public GetUptimeResult(long uptime) {
        this.uptime = uptime;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("uptime", uptime);
        return result;
    }
}

File: packages/device-info/ios/Plugin/Classes/Results/GetUptimeResult.swift
import Foundation
import Capacitor

@objc public class GetUptimeResult: NSObject, Result {
    let uptime: Int

    init(uptime: Int) {
        self.uptime = uptime
    }

    @objc public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["uptime"] = uptime
        return result as AnyObject
    }
}
